clients/src/test/java/org/apache/kafka/common/metrics/MetricsGrpcClient.java

```diff
@@ -0,0 +1,154 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.metrics;
+
+import io.opentelemetry.proto.collector.metrics.v1.ExportMetricsServiceRequest;
+import io.opentelemetry.proto.collector.metrics.v1.ExportMetricsServiceResponse;
+import io.opentelemetry.proto.collector.metrics.v1.MetricsServiceGrpc;
+import io.opentelemetry.proto.metrics.v1.ResourceMetrics;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.stub.StreamObserver;
+import io.grpc.ConnectivityState;
+import io.grpc.ManagedChannel;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.kafka.common.utils.Utils;
+import org.slf4j.LoggerFactory;
+import org.slf4j.Logger;
+
+import java.net.UnknownHostException;
+import java.util.concurrent.TimeUnit;
+import java.net.InetAddress;
+import java.util.List;
+
+
+public class MetricsGrpcClient implements AutoCloseable {
+    private static final Logger log = LoggerFactory.getLogger(MetricsGrpcClient.class);
+
+    public final StreamObserver<ExportMetricsServiceResponse> streamObserver =
+            new StreamObserver<ExportMetricsServiceResponse>() {
+
+        @Override
+        public void onNext(ExportMetricsServiceResponse value) {
+            // Do nothing since response is blank
+        }
+
+        @Override
+        public void onError(Throwable t) {
+            log.debug("Unable to export metrics request to {}. gRPC Connectivity in: {}:  {}",
+                    endpoint, getChannelState(), t.getCause().getMessage());
+        }
+
+        @Override
+        public void onCompleted() {
+            log.info("Successfully exported metrics request to {}", endpoint);
+        }
+    };
+
+    public static final String GRPC_HOST_CONFIG = "OTEL_EXPORTER_OTLP_ENDPOINT";
+
+    public static final int DEFAULT_GRPC_PORT = 4317;
+
+    public static final int GRPC_CHANNEL_TIMEOUT = 30;
+
+    private MetricsServiceGrpc.MetricsServiceStub grpcClient;
+
+    private ManagedChannel grpcChannel;
+
+    private String endpoint;
+
+
+    /**
+     * Start a {@link ManagedChannel} and a {@link MetricsServiceGrpc} at an endpoint defined by the
+     * {@value GRPC_HOST_CONFIG} environment variable
+     */
+    public void initialize() {
+        if (grpcChannel == null) {
+            String grpcHost = System.getenv(GRPC_HOST_CONFIG);
+
+            if (Utils.isBlank(grpcHost)) {
+                log.info("environment variable {} is not set", GRPC_HOST_CONFIG);
```

Comment:
Maybe instead of—or in addition to—this log about the environment variable not being set, you can include a log after the local host address is determined to help the operator see what host _is_ being used for the gRPC host name.

Reply:
I kind of already do it when I say what [endpoint it's connecting to here](https://github.com/confluentinc/kafka/pull/779/files#diff-56704fc6d7e31d947322a6b2f1cf8a0be4fb8f728c598c5cd7b83604cad9f5e8R99)

Reply:
Oh, OK 😄 Sorry, I didn't catch that.